### PR TITLE
 refactor kandinsky img2img for ddpm

### PR DESCRIPTION
**Pull request by yiyixuxu**
this PR refactor the img2img so that it's compatible with ddpm scheduler 

```python
from diffusers import KandinskyImg2ImgPipeline, KandinskyPriorPipeline, DDPMScheduler
from PIL import Image

import torch

REPO_PRIOR = "kandinsky-community/kandinsky-2-1-prior"
REPO = "kandinsky-community/kandinsky-2-1"

model_dtype = torch.float16

prompt = "A red cartoon frog, 4k"

# create prior 
pipe_prior = KandinskyPriorPipeline.from_pretrained(REPO_PRIOR, torch_dtype=model_dtype)
pipe_prior.to("cuda")

# create img2img pipeline
scheduler = DDPMScheduler.from_pretrained(REPO, subfolder="ddpm_scheduler")
pipe = KandinskyImg2ImgPipeline.from_pretrained(REPO, scheduler=scheduler, torch_dtype=model_dtype)
pipe.to("cuda")

# use prior to generate image_emb based on our prompt
generator = torch.Generator(device='cuda').manual_seed(0)
image_emb, zero_image_emb = pipe_prior(prompt, num_inference_steps=25, generator=generator,).to_tuple()


init_image = Image.open("frog.png")
generator = torch.Generator(device='cuda').manual_seed(0)

out = pipe(
    prompt=prompt, 
    image=init_image, 
    height=768, 
    width=768, 
    num_inference_steps=100, 
    generator=generator, 
    image_embeds=image_emb, 
    negative_image_embeds=zero_image_emb, 
    strength=0.2,) 

out.images[0].save("frog_out.png")
```
yiyi_test_img2img_ddpm_out

### CODE REVIEW

**Review by patrickvonplaten — APPROVED** [review]
Let's maybe add a test here as well no?

**Comment by patrickvonplaten** [comment]
@yiyixuxu let's try to get this in :-)

**Comment by patrickvonplaten** [comment]
@yiyixuxu is this still open?
